htdocs/luci-static/resources/homeproxy.js

```diff
@@ -176,6 +176,24 @@ return baseclass.extend({
 		);
 	},
 
+	generateRand: function(type, length) {
+		['base64', 'hex'].includes(type)
+			var byteArr = crypto.getRandomValues(new Uint8Array(length));
+		switch (type) {
+			case 'base64':
+				/* Thanks to https://stackoverflow.com/questions/9267899 */
+				return btoa(String.fromCharCode.apply(null, byteArr));
+			case 'hex':
+				return Array.from(byteArr, (byte) => {
+					return (byte & 255).toString(16).padStart(2, '0');
+				}).join('');
+			case 'uuid':
+				return crypto.randomUUID();
```

Comment:
```suggestion
				return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, (c) =>
					(c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
				);
```

`TypeError: crypto.randomUUID is not a function`

remove `generateUUIDv4` since it's merged into `generateRand`

Reply:
It looks like `crypto.randomUUID()` is only available when the web page uses `https:`.

Reply:
Yep